src/examples/table/TableSelectRows.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import { Body, Cell, Head, HeaderCell, HeaderRow, Row, Table } from '@zendeskgarden/react-tables';
+import { KEY_CODES } from '@zendeskgarden/container-utilities';
+import { Field, Checkbox, Label } from '@zendeskgarden/react-forms';
+
+interface IRowData {
+  index: string;
+  fruit: string;
+  sun: string;
+  soil: string;
+  selected: boolean;
+}
+
+const rowData: IRowData[] = [];
+
+for (let x = 0; x < 10; x++) {
+  rowData.push({
+    index: `row-${x}`,
+    fruit: `Fruit #${x}`,
+    sun: 'Full sun',
+    soil: 'Well draining',
+    selected: false
+  });
+}
+
+const isSelectAllIndeterminate = (rows: IRowData[]) => {
+  const numSelectedRows = rows.reduce((accumulator, row) => {
+    if (row.selected) {
+      return accumulator + 1;
+    }
+
+    return accumulator;
+  }, 0);
+
+  return numSelectedRows > 0 && numSelectedRows < rows.length;
+};
+
+const isSelectAllChecked = (rows: any) => {
+  return rows.every((row: any) => row.selected);
+};
+
+const Example = () => {
+  const [data, setData] = useState(rowData);
+  const [shiftEnabled, setShiftEnabled] = useState(false);
+  const [focusedRowIndex, setFocusedRowIndex] = useState<number | undefined>(undefined);
+
+  return (
+    <div style={{ minWidth: 500 }}>
+      <Table>
+        <Head>
+          <HeaderRow>
+            <HeaderCell isMinimum>
+              <Field>
+                <Checkbox
+                  indeterminate={isSelectAllIndeterminate(data)}
+                  checked={isSelectAllChecked(data)}
+                  onChange={e => {
+                    if (e.target.checked) {
+                      const updatedRows = data.map((row: any) => ({ ...row, selected: true }));
+
+                      setData(updatedRows);
+                    } else {
+                      const updatedRows = data.map((row: any) => ({ ...row, selected: false }));
```

Comment:
TypeScript uses type inference to determine the type for `row`. It seems safe to remove `any` type annotation.